app/services/notification_service.rb

```diff
@@ -56,6 +56,20 @@ def notify(object, recipient_user_ids)
     notification_types(object).each {|type| type.notify(object, recipient_user_ids) }
   end
 
+  def read_all_involving(person)
+    Notification
+      .for(@user)
+      .joins(:notification_actors)
+      .where(
+        notification_actors: {person: person},
```

Comment:
While thinking about this a bit more, another potential problem came to my mind. If multiple people comment on the same post, we aggregate the notifications (that's why it can have multiple actors). So there could be a notification "Person A, Person B and Person C commented on your post", and I think with the current logic, if I block Person B, it would mark this notification as read, and I might miss the comments of Person A and C.

As a user, my expected behavior would be, that it only marks the notification as read, where the blocked user is the only actor.

However this maybe also is only an edge case, when I decide to block somebody who commented on a lot of posts where I also have a lot of notifications from other users I didn't read yet. So maybe it's fine to ignore that in the first version and improve later. But if you find an easy way to select only notifications where there is only one actor (the blocked person), that would be cool to add.

Reply:
I agree, thank you! Also thank you for the explanation on the purpose of storing notification actors!

I implemented only fetching notifications that have exclusively the blocked person as their actor. Let me know what you think!